### PR TITLE
refactor: environment variables in examples dir

### LINKED ISSUE #58: Update ZK bits after merging changes in nightfall_3
See "blocking" PRs for more information.

### LINKED ISSUE #59: Review environment variables
### [Node.js Everywhere with Environment Variables!](https://medium.com/the-node-js-collection/making-your-node-js-work-everywhere-with-environment-variables-2da8cdf6e786)

_Environment variables (env vars) allow applications to behave differently based on the environment you want them to run in._

**Why use them?** Because they externalise all environment specific aspects of your app and keep your app encapsulated.

**When?** Any place in your code that will change based on the environment.

These variables live in an `.env` file. Always "gitignore" this file. You can install VSCODE dotenv extension to benefit from syntax highlight.

In the code, we can handle env vars using the `dotenv` package.

How do we retrieve them? Do you do this everywhere you need a variable (and you may need them in multiple places)? Or should you gather all the variables in one place? 2 options:

- Manually setting and exporting them in a module - recommended in article
- Using a library

**Alternative**

We don’t need `dotenv` to be a runtime dependency. The package provides an option to pre-load variables outside the code. Pre-load the variables and eliminate the code that reads the .env file from the code. You can achieve this by using the `-require (-r)` command.

```bash
node -r dotenv/config server.js
```

This is useful when you want your app to run somewhere where the file does not (and maybe should not) exist, such as in a running docker container or a cloud server.

I highly recommend putting your commands into an npm script. Use the npm scripts outline in VSCode to  run scripts easily.

Share your env vars via `.env.example`.

### More references

- Managing vars with `env-cmd`, creating a file/environment, adding npm scripts: [How to Create Multiple Environments In Node.js 2022](https://www.ultimateakash.com/blog-details/Ii0jLGAKYAo=/How-to-Create-Multiple-Environments-In-Node.js-2022)

- Class Env for managing +1 environaments: [Loading Multiple .env Files for Different Environments in NodeJS](https://thekenyandev.com/blog/environment-variables-strategy-for-node/)

- Jest and environment variables: https://stackoverflow.com/questions/50259025/using-env-files-for-unit-testing-with-jest

### Thoughts

- SDK library is like black box
- App consuming SDK (a.k.a. client) is the place where env vars belong, so each client has to develop a solution for managing these vars
- This means env vars in SDK repo should only live in `/examples` dir, where we can provide suggestions re env vars usage (and their default values)
- Remaining questions a) logging #23 b) what to do with `/__tests__`: tests will need to use env vars too, but this clashes with the black box approach...

### PR DESCRIPTION
**Pull request by imagobea**
Resolves #58 
Resolves #59 

**Conclusions**

- Env vars belong to the application consuming SDK
- Therefore, we only provide an example of how to use them, and as such usage scope is limited to: `/examples` + `__tests__` + `logger`
- As a result I have renamed all env vars, replacing old prefix SDK_ with `APP_`, note also the change to `APP_TOKEN_ADDRESS_1`

The idea is to have an `.env` file per environment, and preloading this from package.json using [dotenv/config](https://www.npmjs.com/package/dotenv).

E.g this is `.env.ganache`

```bash
APP_BLOCKCHAIN_WEBSOCKET_URL=ws://localhost:8546
APP_CLIENT_API_URL=http://localhost:8080
# Nightfall_3 User 1
APP_ETH_PRIVATE_KEY=0x4775af73d6dc84a0ae76f8726bda4b9ecf187c377229cb39e1afa7a18236a69e
APP_NIGHTFALL_MNEMONIC=
# Nightfall_3 ERC20Mock contract address in ganache
APP_TOKEN_ADDRESS_1=0x9b7bD670D87C3Dd5C808ba627c75ba7E88aD066f
```

PR also contributes to #27 

- Have updated scripts based on new approach (not too sure about finalise withdrawal example)
- Have added supporting text

@velkoskis Pls see the text below, we might put something like this beafor each script (?) :thinking: Let me know

> Environment variables (env vars) belong to the application. The SDK does not use env vars except for the `__tests__` dir + logger. On the other hand, the SDK requires data which is environment dependant and that you should store privately, so we provide an example of how to use env vars and SDK.
> 
> The example scripts consume `appConfig` which sources env vars.
> Run the example scripts with the commands provided in the `scripts` section of the `package.json`.
> Note that the scripts load an env file, and that we suggest to use different files per environment. This means you may create an `examples./env.ganache`, based on `.env.example`.

QUESTION: wondering if we could commit the `.env.ganache` since the data belongs to `nightfall_3`.

### CODE REVIEW

**Review by velkoskis — APPROVED** [review]

File: examples/txDeposit.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    const tokenStandard = "ERC20";
+    const value = "0.0001";
+    const txReceipts = await user.makeDeposit({
+      tokenAddress,
+      tokenStandard,
+      value,
+    });
+    console.log("Transaction receipts", txReceipts);
+
+    // # 5 [OPTIONAL] You can check the transaction hash
```
Comment (israelboudoux) [inline examples/txDeposit.ts:33]: I think you forgot some instruction here :)

File: examples/txTransfer.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+      value,
+      recipientAddress: userRecipient.zkpKeys.compressedZkpPublicKey,
+    });
+    console.log("Transaction receipts", txReceipts);
+
+    // # 4 [OPTIONAL] You can check the transaction hash
+    // COMING SOON
+
+    // # 5 [OPTIONAL] You can check transfers that are not yet in a block
+    const pendingTransfers = await userSender.checkLayer2PendingSpentBalances(
```
Comment (israelboudoux) [inline examples/txTransfer.ts:39]: Do you think that `checkLayer2PendingSpentBalances` is a good name? maybe it is, I'm just wondering if you could enhance it.

**Review by israelboudoux — APPROVED** [review]
I've added some comments, but LGTM

File: examples/txTransfer.ts (hunk 2 above)
Comment (imagobea) [reply examples/txTransfer.ts:39]: As of PR #75 we will use `checkPendingTransfers` !! At least more consistent with the other methods

File: examples/txDeposit.ts (hunk 1 above)
Comment (imagobea) [reply examples/txDeposit.ts:33]: It's in L34 but perhaps will be clearer if we assign first to `const`, thank you!

**Comment by velkoskis** [comment]
@imagobea @luizoamorim  What are we going to do with `APP_BLOCKCHAIN_WEBSOCKET_URL` for Goerli testnet and mainnet?
Should we provide a link/short text stating that you should add your own websocket via Alchemy/Infura?

**Pull request merged**